airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -656,6 +671,10 @@ def run_query(self,
                 configuration['query'][
                     'schemaUpdateOptions'] = schema_update_options
 
+        if 'configuration' in api_resource_configs:
```

Comment:
Why require an extra `configuration` key? What other keys would be present?

Reply:
I thought about this param like a possibility to set any settings in one dict, that supported by API (because of 'src_fmt_configs'). But if we say what it's only the way to set 'configuration' and we will not available this way to send any other params, for example, 'jobReference' or 'statistics' .. so way what I could define one dict with all params and send it. So if it's not needed and it will be only for 'configuration' - it's of course not needed. What do you think? Use it only for 'configuration' ?

Reply:
Good point, `jobReference` could be useful to set, as that is how you can specify an explicit `location` for the job to run.